### PR TITLE
Change strategy about inactive related logic

### PR DESCRIPTION
**Pull request by raindust**

### CODE REVIEW

File: blockchain/txvalidator.go [hunk 1]
```diff
@@ -84,14 +83,23 @@ func (b *BlockChain) CheckTransactionSanity(blockHeight uint32, txn *Transaction
...
 func (b *BlockChain) CheckTransactionContext(blockHeight uint32, txn *Transaction) ErrCode {
 	// check if duplicated with transaction in ledger
 	if exist := b.db.IsTxHashDuplicate(txn.Hash()); exist {
 		log.Warn("[CheckTransactionContext] duplicate transaction check failed.")
 		return ErrTransactionDuplicate
 	}
 
+	// check to ensure only producer related tx existing in current block
+	// when inactive mode is on
+	if DefaultLedger.Arbitrators.IsInactiveMode() && !txn.IsProducerRelatedTx() &&
```
Comment (alexpan0610) [inline blockchain/txvalidator.go]: Are InactiveArbitrators and IllegalType not a ProducerRelated transaction?
The comments saying only producer related tx is allowed, shouldn't it be
```go
if DefaultLedger.Arbitrators.IsInactiveMode() && !txn.IsProducerRelatedTx() {
    ...
}
```

File: core/types/transaction.go [hunk 2]
```diff
@@ -42,6 +42,8 @@ const (
 	IllegalBlockEvidence     TxType = 0x10
 	IllegalSidechainEvidence TxType = 0x11
 	InactiveArbitrators      TxType = 0x12
+
```
Comment (alexpan0610) [inline core/types/transaction.go]: Remove the blank line.

File: dpos/network.go [hunk 3]
```diff
@@ -298,14 +309,20 @@ func NewDposNetwork(account account.Account, medianTime dtime.MedianTimeSource,
 	listener manager.NetworkEventListener) (*network, error) {
 	network := &network{
 		listener:                 listener,
+		proposalDispatcher:       nil,
```
Comment (alexpan0610) [inline dpos/network.go]: Remove 312~316 lines.

File: pow/service.go [hunk 4]
```diff
@@ -161,8 +161,7 @@ func (pow *Service) AssignCoinbaseTxRewards(block *types.Block, totalReward comm
 		rewardDposArbiter := common.Fixed64(math.Ceil(float64(totalReward) * 0.35))
 		rewardMergeMiner := common.Fixed64(totalReward) - rewardCyberRepublic - rewardDposArbiter
 
-		if rewards := pow.arbiters.GetArbitersRoundReward();
-			rewards != nil && len(rewards) > 0 {
+		if rewards := pow.arbiters.GetArbitersRoundReward(); rewards != nil && len(rewards) > 0 {
```
Comment (alexpan0610) [inline pow/service.go]: Only `len(rewards) > 0` is ok, because `len(nil) == 0`.

File: pow/service.go [hunk 5]
```diff
@@ -238,11 +237,13 @@ func (pow *Service) GenerateBlock(minerAddr string) (*types.Block, error) {
 	txs := pow.txMemPool.GetTxsInPool()
 	sort.Slice(txs, func(i, j int) bool {
```
Comment (alexpan0610) [inline pow/service.go]: May be a better solution.
```go
	isHighPriority := func(tx *Transaction) bool {
		if tx.IsIllegalTypeTx() || tx.IsInactiveArbitrators() ||
			tx.IsSideChainPowTx() tx.IsUpdateVersion() ||
			tx.IsActivateProducerTx() {
			return true
		}
		return false
	}

	sort.Slice(txs, func(i, j int) bool {
		if isHighPriority(txs[i]) {
			return true
		}
		if isHighPriority(txs[j]) {
			return false
		}
		return txs[i].FeePerKB > txs[j].FeePerKB
	})
```

File: blockchain/txvalidator.go [hunk 6]
```diff
@@ -116,18 +124,32 @@ func (b *BlockChain) CheckTransactionContext(blockHeight uint32, txn *Transactio
 		if err := b.checkIllegalBlocksTransaction(txn); err != nil {
 			log.Warn("[CheckIllegalBlocksTransaction],", err)
 			return ErrTransactionPayload
+		} else {
```
Comment (alexpan0610) [inline blockchain/txvalidator.go]: No need of else switch, just `return Success`.

File: blockchain/txvalidator.go [hunk 7]
```diff
@@ -116,18 +124,32 @@ func (b *BlockChain) CheckTransactionContext(blockHeight uint32, txn *Transactio
...
+			return Success
 		}
 
 	case IllegalSidechainEvidence:
 		if err := b.checkSidechainIllegalEvidenceTransaction(txn); err != nil {
 			log.Warn("[CheckSidechainIllegalEvidenceTransaction],", err)
 			return ErrTransactionPayload
+		} else {
```
Comment (alexpan0610) [inline blockchain/txvalidator.go]: No need of else switch, just `return Success`.

File: blockchain/txvalidator.go [hunk 8]
```diff
@@ -116,18 +124,32 @@ func (b *BlockChain) CheckTransactionContext(blockHeight uint32, txn *Transactio
...
+			return Success
 		}
 
 	case InactiveArbitrators:
 		if err := b.checkInactiveArbitratorsTransaction(txn); err != nil {
 			log.Warn("[CheckInactiveArbitrators],", err)
 			return ErrTransactionPayload
+		} else {
```
Comment (alexpan0610) [inline blockchain/txvalidator.go]: No need of else switch, just `return Success`.

File: blockchain/txvalidator.go [hunk 9]
```diff
@@ -116,18 +124,32 @@ func (b *BlockChain) CheckTransactionContext(blockHeight uint32, txn *Transactio
...
+			return Success
+		}
+
+	case UpdateVersion:
+		if err := b.checkUpdateVersionTransaction(txn); err != nil {
+			log.Warn("[checkUpdateVersionTransaction],", err)
+			return ErrTransactionPayload
+		} else {
```
Comment (alexpan0610) [inline blockchain/txvalidator.go]: No need of else switch, just `return Success`.

File: blockchain/txvalidator.go (hunk 1 above)
Comment (raindust) [reply blockchain/txvalidator.go]: this comment should be "producer related tx and DPOS related tx", changed it.

File: core/types/transaction.go (hunk 2 above)
Comment (raindust) [reply core/types/transaction.go]: ok, fixed it.

File: dpos/network.go (hunk 3 above)
Comment (raindust) [reply dpos/network.go]: ok, fixed it.

File: pow/service.go (hunk 4 above)
Comment (raindust) [reply pow/service.go]: Thanks, learned a lot.

File: pow/service.go (hunk 5 above)
Comment (raindust) [reply pow/service.go]: Yes, that's a good refactor practice, good job.

File: blockchain/txvalidator.go (hunk 6 above)
Comment (raindust) [reply blockchain/txvalidator.go]: Maybe someone thinks explicitly write if-else condition will be easier to read even if there are only returns expression. Well, there is no certain answer about this, modified it to uniform out code style.

File: blockchain/txvalidator.go (hunk 7 above)
Comment (raindust) [reply blockchain/txvalidator.go]: Ok, fixed it.

File: blockchain/txvalidator.go (hunk 8 above)
Comment (raindust) [reply blockchain/txvalidator.go]: Ok, fixed it.

File: blockchain/txvalidator.go (hunk 9 above)
Comment (raindust) [reply blockchain/txvalidator.go]: Ok, fixed it.

**Pull request merged**